### PR TITLE
Feature/model protocols

### PR DESCRIPTION
**Pull request by vadymmarkov**
@zenangst @RamonGilabert 
1. Use protocols for models.
2. Refactor code.
3. Update demos according to changes and move `Demos` to `Example`.

Should fix https://github.com/hyperoslo/Wall/issues/71 and https://github.com/hyperoslo/Wall/issues/65

### CODE REVIEW

**Comment by RamonGilabert** [comment]
Looks way much cleaner now, bro, awesome job! :) let's see if @zenangst takes a look at it! :)

File: Example/App.Net/App.Net/Post.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  var publishDate: NSDate?
+  var text: String?
+  var liked = false
+  var seen = false
+  var likeCount = 0
+  var seenCount = 0
+  var commentCount = 0
+  var author: Profileable?
+  var group: Groupable?
+  var location: Location?
```
Comment (zenangst) [inline Example/App.Net/App.Net/Post.swift]: Shouldn't this also get a suitable protocol name?

Comment (vadymmarkov) [inline Example/App.Net/App.Net/Post.swift]: @zenangst I thought about `Location` as a simple struct that holds name and `Coordinate`. Does it make sense to convert it to protocol?

Comment (zenangst) [inline Example/App.Net/App.Net/Post.swift]: @vadymmarkov Good point! :sunglasses:

**Comment by zenangst** [comment]
unknown-4

**Pull request merged**